Musico/app/models/user.rb

```diff
@@ -1,5 +1,30 @@
 class User < ApplicationRecord
-    has_and_belongs_to_many :songs
     validates :name, :email, :password, :dob, presence: :true
+
+    #follow
+    has_many :follower_follows, foreign_key: :following_id, class_name: "Follow" , dependent: :destroy
+    has_many :followers, through: :follower_follows, source: :follower
+    has_many :following_follows, foreign_key: :follower_id, class_name: "Follow", dependent: :destroy
+    has_many :followings, through: :following_follows, source: :following
+
+    #like
+    has_many :likes , dependent: :destroy
+
+    #history
+    has_many :histories , dependent: :destroy
+    
+    #songs
+    has_and_belongs_to_many :songs, dependent: :destroy
+   
+    #song_queue
+    has_one :song_queue ,  dependent: :destroy
+
+    #group_sessions
+    has_and_belongs_to_many :group_sessions , dependent: :destroy
+    has_many :group_sessions , dependent: :destroy 
```

Comment:
These two method will conflict, change the name of one of them, make relevant to the context